cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -77,7 +77,7 @@ func NewCMServer() *CMServer {
 		NodeMilliCPU:            1000,
 		NodeMemory:              resource.MustParse("3Gi"),
 		SyncNodeList:            true,
-		SyncNodeStatus:          false,
+		SyncNodeStatus:          true,
```

Comment:
It's required for this to work. In order to prevent scheduling to nodes that can not run pods there needs to be a constant monitoring and updating of all node statuses. If the node goes down, gets rebooted, or even re-purposed then that node isn't available to run pods. The only way that I see in the current implementation that is noticed is with the node controller pinging the node and updating the status for the scheduler.

I'm not sure why you would ever want this off to be honest.